pkg/agent/handler/credential.go

```diff
@@ -98,30 +100,43 @@ func (h *credentials) Handle(action proto.Event_Type, _ *proto.EventMeta, resour
 		if err != nil {
```

Comment:
We are probably missing to set the Error status on the credential resource, If the key is invalid (invalid PEM content) the newPub() is going to fail and return error however the status on the resource is not changed because the flow returns from this block, so it will remaining in Pending status.

Reply:
made changes to catch this error and build a failed status

Reply:
We need to handle the failed status in the other handlers too

Reply:
We will want to mark them as TODOs or create stories